lib/common/domain/check_if_app_terms_should_be_shown_usecase.dart

```diff
@@ -0,0 +1,27 @@
+import 'package:zachranobed/services/auth_service.dart';
+import 'package:zachranobed/features/appConfiguration/data/get_last_app_terms_version_use_case.dart';
+
+class CheckIfAppTermsShouldBeShownUseCase {
+  final AuthService authService;
```

Comment:
`AuthService` is an implementation, which should not be in domain

Also you are using a repo from "data" layer directly and not apply a dependency inversion

For now I would keep it as-is, but create an issue to refactor it later (create some kind of user repository interface & implementation), ok?

Reply:
This is ok by me, thank you for noticing - I created a technical debt task and linked in in TODO